core/state_transition_ovm.go

```diff
@@ -0,0 +1,190 @@
+package core
+
+import (
+	"bytes"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/core/vm"
+)
+
+var GodAddress = common.HexToAddress("0x444400000000000000000000000000000000000")
+var ZeroAddress = common.HexToAddress("0x0000000000000000000000000000000000000000")
+
+type ovmTransaction struct {
+	Timestamp     *big.Int       "json:\"timestamp\""
+	BlockNumber   *big.Int       "json:\"blockNumber\""
+	L1QueueOrigin uint8          "json:\"l1QueueOrigin\""
+	L1TxOrigin    common.Address "json:\"l1TxOrigin\""
+	Entrypoint    common.Address "json:\"entrypoint\""
+	GasLimit      *big.Int       "json:\"gasLimit\""
+	Data          []uint8        "json:\"data\""
+}
+
+func toExecutionManagerRun(evm *vm.EVM, msg Message) (Message, error) {
+	tx := ovmTransaction{
+		evm.Context.Time,
+		evm.Context.BlockNumber, // TODO (what's the correct block number?)
+		uint8(msg.QueueOrigin().Uint64()),
+		*msg.L1MessageSender(),
+		*msg.To(),
+		big.NewInt(int64(msg.Gas())),
+		msg.Data(),
+	}
+
+	var abi = vm.OvmExecutionManager.ABI
+	var args = []interface{}{
+		tx,
+		vm.OvmStateManager.Address,
+	}
+
+	ret, err := abi.Pack("run", args...)
+	if err != nil {
+		return nil, err
+	}
+
+	outputmsg, err := modMessage(
+		msg,
+		msg.From(),
+		&vm.OvmExecutionManager.Address,
+		ret,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return outputmsg, nil
+}
+
+func asOvmMessage(tx *types.Transaction, signer types.Signer) (Message, error) {
+	msg, err := tx.AsMessage(signer)
+	if err != nil {
+		return msg, err
+	}
+
+	if msg.From() == GodAddress {
+		return msg, nil
+	}
+
+	v, r, s := tx.RawSignatureValues()
+	v = big.NewInt(int64(v.Uint64() - 35 - 2*420))
```

Comment:
```suggestion
	v = big.NewInt(int64(v.Uint64() - 35 - 2* CHAIN_ID))
```

Could refactor this out to a CHAIN_ID constant, maybe also add a comment that this is EIP-155?